cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -155,12 +155,15 @@ func newNodeAnnotator(client clientset.Interface, nodeInformer coreinformers.Nod
 						return false
 					}
 
-					err = mergeManagedTaints(node, desiredTaints)
-					if err != nil {
-						klog.Errorf("Error merging taints: %v", err)
-						return false
+					if node.ObjectMeta.Annotations == nil {
+						node.ObjectMeta.Annotations = make(map[string]string)
 					}
 
+					// Set last applied taints annotation. Cluster autoscaler relies on
+					// this.
+					// Note: We don't merge the taints like we do for labels because we
+					// don't want to reapply any taints the user might have deleted.
```

Comment:
how do you know if they were deleted or they are still valid?

Reply:
is ok to drop existing taints?

Reply:
If the taints weren't deleted, they would still be present on the nodes. So, there is no need to reapply them. Lot of users are relying on applying taints on node creation and removing them once the node is up. They don't want the taint to be reapplied.